### ISSUE TITLE
Auto-generate services files

### ISSUE THREAD

**Issue by FroMage** [quarkusio/quarkus#3363]
We should be able to generate a `META-INF/services/org.foo.Bar` file based on the existence of implementations of `org.foo.Bar`. Possibly requiring an annotation such as `@Service`, but I'm not sure we really need that.

Those files are just very annoying to write because it's easy to mix `service` and `services` and get the name of the file wrong, as well as its contents, because nothing is type-safe.

**Comment by geoand** [comment]
@FroMage I like the idea! But

**Comment by FroMage** [comment]
Right. We could invert the logic, and require service interfaces to have `@Service` except for "well-known interfaces" (those that belong in EE that we can't touch). This way we can look for all subtypes of those service interfaces. It's a bit more expensive, but easier for users since there are less interfaces than implementations.
